x-pack/test/functional/services/machine_learning/custom_urls.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import expect from '@kbn/expect';
+
+import { FtrProviderContext } from '../../ftr_provider_context';
+
+export function MachineLearningCustomUrlsProvider({ getService }: FtrProviderContext) {
+  const testSubjects = getService('testSubjects');
+
+  return {
+    async assertCustomUrlLabelValue(expectedValue: string) {
+      const actualCustomUrlLabel = await testSubjects.getAttribute(
+        'mlJobCustomUrlLabelInput',
+        'value'
+      );
+      expect(actualCustomUrlLabel).to.eql(expectedValue);
+    },
+
+    async setCustomUrlLabel(customUrlsLabel: string) {
+      await testSubjects.setValue('mlJobCustomUrlLabelInput', customUrlsLabel, {
+        clearWithKeyboard: true,
+      });
+      await this.assertCustomUrlLabelValue(customUrlsLabel);
+    },
+
+    async assertCustomUrlItem(index: number) {
+      await testSubjects.existOrFail(`mlJobEditCustomUrlInput_${index}`);
+    },
+
+    async addCustomUrl() {
```

Comment:
Since we have a `addCustomUrl` in the job wizard service as well, I would suggest to rename this one to something like `saveCustomUrl` in order to avoid confusion.

Reply:
Renamed in https://github.com/elastic/kibana/pull/52269/commits/5964af0efb796bb88324c2233ff59ec7d45b028b